addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiScanRule.java

```diff
@@ -396,8 +402,12 @@ private void searchForMathsExecution(
                                             + ".*"
                                             + DELIMITER
                                             + "[\\w\\W]*";
+                            Matcher matcher = Pattern.compile(regex).matcher(output);
+                            String strippedTest = renderTest.replaceAll("[^A-Za-z0-9]+", "");
 
-                            if (output.contains(renderResult) && output.matches(regex)) {
+                            if (output.contains(renderResult)
+                                    && matcher.matches()
+                                    && !matcher.group(0).contains(strippedTest)) {
```

Comment:
Extract a method for the new logic, it should only do the extra work when it contains. I'd even move this to the `TemplateFormat` as this is specific to Go expression.

Reply:
Fixed, I think :wink:

Reply:
It would have been better to only compile/match once but since that's only if there's a match and in Go.